.github/workflows/e2e.yml

```diff
@@ -16,7 +16,37 @@ env:
   CURL_HOME: .
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      sources: ${{ steps.filter.outputs.sources }}
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name == 'push'
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            sources:
+              - .github/workflows/e2e.yml
+              - e2e/**
+              - scripts/workflows/e2e-matrix.py
+              - scripts/workflows/provision-darwin.sh
+              - scripts/workflows/provision-linux.sh
+              - scripts/test-uis.py
+              - src/canisters/frontend/**
+              - src/dfx/**
+              - src/dfx-core/**
+              - src/distributed/**
+              - src/lib/**
```

Comment:
Why not `src/**`?

Reply:
In order to put something else in `src/<something else>/**` without triggering e2e